tjpb/magister_bef_may.go

```diff
@@ -0,0 +1,166 @@
+package main
+
+import (
+	"bytes"
+	"encoding/csv"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/dadosjusbr/storage"
+	"github.com/gocarina/gocsv"
+)
+
+// magister_bef_may.go parse all servants.pdf before may/2020.
+
+type magBefMay struct { // Our example struct, you can use "-" to ignore a field
+	Name             string   `csv:"name"`
+	Role             string   `csv:"role"`
+	Workplace        string   `csv:"workplace"`
+	Wage             *float64 `csv:"wage"`
+	PersonalBenefits *float64 `csv:"personalBenefits"`
+	Gratification    *float64 `csv:"gratification"`
+	Perks            *float64 `csv:"perks"`
+	EventualBenefits *float64 `csv:"eventualBenefits"`
+	PrevContribution *float64 `csv:"prevContribution"`
+	IncomeTax        *float64 `csv:"incomeTax"`
+	OthersDisc       *float64 `csv:"othersDisc"`
+	CeilRetention    *float64 `csv:"ceilRetention"`
+	TotalDisc        *float64 `csv:"totalDisc"`
+	TotalIncome      *float64 `csv:"totalIncome"`
+	IncomeFinal      *float64 `csv:"netIncome"`
+	Daily            *float64 `csv:"daily"`
+}
+
+func parserMagBefMay(path string) ([]storage.Employee, error) {
+	// We generate this template using release 1.2.1 of https://github.com/tabulapdf/tabula
+	templateArea := []string{"101.475,35.64,557.865,130.68",
+		"100.485,129.69,564.795,234.63",
+		"104.445,231.66,564.795,745.473",
+		"102.465,365.31,554.895,397.98",
+		"102.465,404.91,554.895,747.45"}
+	csvFinal := headersMag()
+	for i, templ := range templateArea {
+		//This cmd execute a tabula script(https://github.com/tabulapdf/tabula-java)
+		//where tmpl is the template area, which corresponds to the coordinates (x1,2,y1,2) of
+		//one or more columns in the table.
+		cmdList := strings.Split(fmt.Sprintf(`java -jar tabula-1.0.3-jar-with-dependencies.jar -t -a %v -p all %v`, templ, filepath.Base(path)), " ")
+		cmd := exec.Command(cmdList[0], cmdList[1:]...)
+		var outb, errb bytes.Buffer
+		cmd.Stdout = &outb
+		cmd.Stderr = &errb
+		cmd.Run()
+		reader := csv.NewReader(&outb)
+		reader.FieldsPerRecord = -1
+		rows, err := reader.ReadAll()
+		if err != nil {
+			logError("Error reading rows from stdout: %v", err)
+			os.Exit(1)
+		}
+
+		// When the templ refers to worksplace Column, treating double lines is necessary
+		if i == 2 {
+			// Pass rows and a knew invariable and non-empty column pos.
+			rows = treatDoubleLines(rows, 3)
+		}
+		// When the templ refers to column of numbers, treating cels to format numbers and
+		// remove characters.
+		if i == 3 || i == 4 {
+			rows = fixNumberColumns(rows)
+		}
+		file, err := os.Create(fmt.Sprintf("%v.csv", i))
+		if err != nil {
+			logError("Error creating csv: %q", err)
+			os.Exit(1)
+		}
+		defer file.Close()
```

Comment:
`defer` nãso vai ser invocado ao final do bloco, ele vai ser invocado ao final de uma função. Nesse caso, você tem duas opções:

1. cria uma função anônima para executar esse bloco aqui
1. força o `Close` e `Flush` no lugar certo